engine/crates/engine-v2/src/executor/coordinator.rs

```diff
@@ -1,76 +1,106 @@
-use std::sync::Arc;
+use std::{pin::Pin, sync::Arc};
 
 use engine_parser::types::OperationType;
 use futures_locks::Mutex;
-use futures_util::StreamExt;
+use futures_util::{stream::FuturesUnordered, Future, StreamExt};
 
-use super::{Executor, ExecutorRequest, ResponseProxy};
+use super::{Executor, ExecutorError, ExecutorRequest, ResponseProxy};
 use crate::{
-    plan::{ExecutionPlans, PlanId, RequestPlan},
-    response::{OperationSelectionSet, Response},
+    plan::{ExecutionPlans, PlanId, PlannedOperation},
+    request::OperationFields,
+    response::{ReadSelectionSet, Response},
     Engine,
 };
 
 pub struct ExecutorCoordinator<'a> {
     engine: &'a Engine,
     operation_type: OperationType,
-    operation_selection_set: OperationSelectionSet,
+    operation_fields: OperationFields,
+    selection_set: ReadSelectionSet,
     plans: ExecutionPlans,
     response: Arc<Mutex<Response>>,
 }
 
 impl<'a> ExecutorCoordinator<'a> {
-    pub fn new(engine: &'a Engine, request_plan: RequestPlan) -> Self {
+    pub fn new(engine: &'a Engine, planned_operation: PlannedOperation) -> Self {
         Self {
             engine,
-            operation_type: request_plan.operation_type,
-            operation_selection_set: request_plan.operation_selection_set,
-            response: Arc::new(Mutex::new(Response::new(
-                engine.schema.clone(),
-                request_plan.response_fields,
-            ))),
-            plans: request_plan.execution_plans,
+            operation_type: planned_operation.ty,
+            selection_set: planned_operation.selection_set,
+            operation_fields: planned_operation.fields,
+            response: Arc::new(Mutex::new(Response::new(planned_operation.strings))),
+            plans: planned_operation.plans,
         }
     }
 
     pub async fn execute(&mut self) -> serde_json::Value {
-        let (sender, mut receiver) = futures_channel::mpsc::unbounded::<CoordinationTask>();
+        let mut futures = FuturesUnordered::<Pin<Box<dyn Future<Output = CoordinationTask> + Send + 'static>>>::new();
```

Comment:
Is there a reason this can't be `FuturesUnordered::<BoxFuture<'static, CoordinationTask>>`?  Bit easier to read

Reply:
Nop it's exactly the same actually, just didn't think of it

Reply:
will change it